### PR TITLE
[ur] Adds UR event conformance tests

### PR DESCRIPTION
**Pull request by veselypeta**
This PR adds the only UnitUR test for events - I can create additional tasks to create new tests for the reminaing entry-points.

### CODE REVIEW

File: test/conformance/event/urEventWait.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTest::SetUp());
+        ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_WRITE_ONLY, size,
+                                         nullptr, &src_buffer));
+        ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_READ_ONLY, size,
+                                         nullptr, &dst_buffer));
+        input.assign(count, 42);
+        ASSERT_SUCCESS(urEnqueueMemBufferWrite(queue, src_buffer, true, 0, size,
+                                               input.data(), 0, nullptr,
+                                               &event));
+        ASSERT_SUCCESS(urEventWait(1, &event));
```
Comment (aarongreig) [inline test/conformance/event/urEventWait.cpp:17]: do we need to do this? the buffer write is blocking

Comment (veselypeta) [reply test/conformance/event/urEventWait.cpp:17]: Fixed

**Review by aarongreig — APPROVED** [review]
LGTM

**Review by kbenzie — APPROVED** [review]
LGTM

**Pull request merged**